server/controllers/stock/inventory/core.js

```diff
@@ -38,11 +39,55 @@ var errors = {
 exports.getIds = getIds;
 exports.getItemsMetadata = getItemsMetadata;
 exports.getItemsMetadataById = getItemsMetadataById;
+exports.createItemsMetadata = createItemsMetadata;
+exports.updateItemsMetadata = updateItemsMetadata;
 exports.hasBoth = hasBoth;
 exports.errors = errors;
 exports.errorHandler = errorHandler;
 
 /**
+* Create inventory metadata in the database
+*
+* @function createItemsMetadata
+* @return {Promise} Returns a database query promise
+*/
+function createItemsMetadata(record, session) {
+
+  record.enterprise_id = session.enterprise.id;
+  record.uuid = db.bid(record.uuid || uuid.v4());
+  record.group_uuid = db.bid(record.group_uuid);
+
+  let sql = `INSERT INTO inventory SET ?;`;
+  /*
+   * return a promise which can contains result or error which is caught
+   * in the main controller (inventory.js)
+   */
+  return db.exec(sql, [record])
+  .then(() => uuid.unparse(record.uuid));
+}
+
+/**
+* Update inventory metadata in the database
+*
+* @function updateItemsMetadata
+* @return {Promise} Returns a database query promise
+*/
+function updateItemsMetadata(record, identifier) {
+
+  identifier = db.bid(identifier);
+  record.uuid = identifier;
+  record.group_uuid = db.bid(record.group_uuid);
+
+  let sql = `UPDATE inventory SET ? WHERE uuid = ?;`;
+  /*
+   * return a promise which can contains result or error which is caught
```

Comment:
This is a useful comment, but probably should be in the top comment of the function, with the `@return` keyword.